demo_nodes_py/demo_nodes_py/logging/use_logger_service.py

```diff
@@ -0,0 +1,180 @@
+            return [False, None]
+
+        request = GetLoggerLevels.Request()
+        request.names.append(self._remote_node_name)
+
+        future = self._logger_get_client.call_async(request)
+        rclpy.spin_until_future_complete(self, future)
+
+        ret_results = future.result()
+        if not ret_results:
+            return [False, None]
+
+        return [True, ret_results.levels[0].level]
+
+
+def get_logger_level_func(test_node):
+    ret, level = test_node.get_logger_level_on_remote_node()
+    if ret:
+        test_node.get_logger().info('Current logger level: ' + str(level))
+    else:
+        test_node.get_logger().error('Failed to get debug logger level via logger service !')
+
+
+def main(args=None):
+    rclpy.init(args=args)
+
+    logger_service_node = LoggerServiceNode()
+    test_node = TestNode('LoggerServiceNode')
+
+    executor = SingleThreadedExecutor()
+    executor.add_node(logger_service_node)
+
+    def thread_process(node):
+        executor.spin()
+
+    thread = threading.Thread(target=thread_process, args=(executor,))
+    thread.start()
+
+    # Output with default logger level
+    test_node.get_logger().info('Output with default logger level:')
+    msg = String()
+    msg.data = 'Output 1'
+    test_node.pub.publish(msg)
+    time.sleep(0.5)
+
+    # Get logger level. Logger level should be 0 (Unset)
+    get_logger_level_func(test_node)
+
+    # Output with debug logger level
+    test_node.get_logger().info('Output with debug logger level:')
+    if test_node.set_logger_level_on_remote_node(LoggingSeverity.DEBUG):
+        msg = String()
+        msg.data = 'Output 2'
+        test_node.pub.publish(msg)
+        time.sleep(0.5)
+    else:
+        test_node.get_logger().error('Failed to set debug logger level via logger service !')
+
+    # Get logger level. Logger level should be 10 (Debug)
+    get_logger_level_func(test_node)
+
+    # Output with warn logger level
+    test_node.get_logger().info('Output with warn logger level:')
+    if test_node.set_logger_level_on_remote_node(LoggingSeverity.WARN):
+        msg = String()
+        msg.data = 'Output 3'
+        test_node.pub.publish(msg)
+        time.sleep(0.5)
+    else:
+        test_node.get_logger().error('Failed to set debug logger level via logger service !')
+
+    # Get logger level. Logger level should be 30 (warn)
+    get_logger_level_func(test_node)
+
+    # Output with error logger level
+    test_node.get_logger().info('Output with error logger level:')
+    if test_node.set_logger_level_on_remote_node(LoggingSeverity.ERROR):
+        msg = String()
+        msg.data = 'Output 4'
+        test_node.pub.publish(msg)
+        time.sleep(0.5)
+    else:
+        test_node.get_logger().error('Failed to set debug logger level via logger service !')
+
+    # Get logger level. Logger level should be 40 (Error)
+    get_logger_level_func(test_node)
+
+    executor.shutdown()
+    if thread.is_alive():
+        thread.join(3)
```

Comment:
Is there any reason to use a timeout to join? If yes, I think we need to deal with the issue in `rclpy`.

Reply:
Okay, remove it.